plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/checkpoint/CheckpointEntryIterator.java

```diff
@@ -173,6 +176,7 @@ public CheckpointEntryIterator(
             boolean checkpointRowStatisticsWritingEnabled,
             int domainCompactionThreshold,
             TupleDomain<DeltaLakeColumnHandle> partitionConstraint,
+            TupleDomain<DeltaLakeColumnHandle> nonPartitionConstraint,
```

Comment:
I don't think there should be a need to send partitioned and un-partitioned columns TupleDomain separately.
The caller should send one combined TupleDomain.
`io.trino.plugin.deltalake.util.DeltaLakeDomains#partitionMatchesPredicate` should be rewritten to iterate over `partitionKeys` instead and not be affected by any extra Domains passed to it.

Reply:
I saw the initial of the `DeltaLakeTableHandle` has two interface `getEnforcedPartitionConstraint` and `getNonPartitionConstraint`. potentially, I think we may need to modify this two interface, but that involves two many interface/caller changes. For Now, I just update the `CheckpointEntryIterator`, are we cool on this?

Reply:
Revisit. Remove the most parts that extend the to add the `nonPartitionConstraint`, only keep the one place in `TransactionLogAccess#getActiveFiles` and it's caller to pass the value. It's looks much cleaner.